app/Http/QueryBuilder/Scraper/ScraperHandler.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Http\QueryBuilder\Scraper;
+
+use App\Providers\SerializerFactory;
+
+class ScraperHandler
+{
+    /**
+     * @var mixed
+     */
+    private mixed $response;
+
+    /**
+     * @param string $getterFuncName
+     * @param string $requestResolverClassName
+     * @param QueryResolver $queryResolver
+     */
+    public function __construct(string $getterFuncName, string $requestResolverClassName, QueryResolver $queryResolver)
+    {
+        $getterFuncName = "get".ucfirst(strtolower($getterFuncName));
+
+        $this->response = app('JikanParser')
+            ->$getterFuncName(
+                new $requestResolverClassName(
+                    ...$queryResolver->getQueryValuesAsArray()
+                )
+            );
+    }
+
+
+    /**
+     * @return array
+     */
+    public function getSerializedJSON(): array
+    {
+        return \json_decode(
+            SerializerFactory::createV4()
```

Comment:
This can be resolved through the service container too.

Of course this would require some extra logic in the `AppServiceProvider` class:
https://github.com/jikan-me/jikan-rest/blob/master/app/Providers/AppServiceProvider.php#L122

```php
$this->app->singleton(Serializer::class, function ($app) {
    return SerializerFactory::createV4();
});
```

But if there is a need for a different serializer instance, then you might want to tag the instances in the container or use interfaces. In case of tagging you have to explicitly register those classes which have a dependency on a serializer which is not the same as the default one.
https://laravel.com/docs/9.x/container#tagging

Reply:
@pushrbx I'm not sure how this would be done. I've simply added this into `Helpers/ScraperHelper` which is called in the controller.